polaris/dataset/_base.py

```diff
@@ -362,3 +355,35 @@ def __repr__(self):
 
     def __str__(self):
         return self.__repr__()
+
+    def _cache_zarr(self, destination: str | PathLike, if_exists: ZarrConflictResolution):
+        """
+        Caches the Zarr archive to a local directory.
+        """
+        from polaris.hub.storage import S3Store
+
+        if not self.uses_zarr:
+            return
+
+        current_zarr_root = Path(self.zarr_root_path)
```

Comment:
It's true that here, `PurePath` would work because it's just stitching paths together. But according to the doc [`Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path), when instantiated, will be either a `PosixPath` or a `WindowsPath`, according to the platform the user is running on.

So for platform-agnostic file operations, `Path` is appropriate object to use.

There's a few places where we used [`str.split`](https://docs.python.org/3.10/library/stdtypes.html#str.split), with an assumption that `/` would be the path separator. That breaks on Windows.